packages/solid/src/elements/index.ts

```diff
@@ -0,0 +1,142 @@
+import type {
+  ASCIIFontOptions,
+  BoxOptions,
+  InputRenderableOptions,
+  Renderable,
+  RenderableOptions,
+  SelectOption,
+  SelectRenderableOptions,
+  TabSelectRenderableOptions,
+  TextOptions,
+} from "@opentui/core";
+import {
+  ASCIIFontRenderable,
+  BoxRenderable,
+  GroupRenderable,
+  InputRenderable,
+  SelectRenderable,
+  TabSelectRenderable,
+  TextRenderable,
+} from "@opentui/core";
+import { onCleanup, children as resolveChildren, type JSX, type Ref } from "solid-js";
+import { createElement, effect, insert, setProp, spread, use } from "../reconciler";
+export * from "./hooks";
+
+export const elements = {
+  opentui_box: BoxRenderable,
+  opentui_group: GroupRenderable,
+  opentui_input: InputRenderable,
+  opentui_tab_select: TabSelectRenderable,
+  opentui_text: TextRenderable,
+  opentui_select: SelectRenderable,
+  opentui_ascii_font: ASCIIFontRenderable,
+};
+
+export type Element = keyof typeof elements;
+
+type RenderableNonStyleKeys = "buffered";
+
+type ElementProps<
+  T extends RenderableOptions,
+  K extends Renderable = Renderable,
+  NonStyleKeys extends keyof T = RenderableNonStyleKeys,
+> = {
+  style?: Omit<T, NonStyleKeys | RenderableNonStyleKeys>;
+  ref?: Ref<K>;
+} & T;
+
+const createCustomElement = <T extends Record<string, any>>(tagName: string, acceptChildren?: string | true) => {
+  return (props: T) => {
+    const element = createElement(tagName);
+    onCleanup(() => {
+      element.destroy();
+    });
+    if (props.ref) {
+      typeof props.ref === "function"
+        ? use(props.ref, element, undefined)
+        : // @ts-expect-error: directly assigning to ref as is done after transpile
+          (props.ref = element);
+    }
+    spread(element, props, true);
+    if (acceptChildren) {
+      const resolved = resolveChildren(() => props.children);
+      if (acceptChildren === true) {
+        insert(element, resolved);
+      } else {
+        effect((prev) => {
+          const value = resolved();
+          if (prev !== value) {
+            setProp(element, acceptChildren, value, prev);
+          }
+
+          return value;
+        }, "");
+      }
+    }
+
+    return element as JSX.Element;
+  };
+};
+
+type ContianerProps = { children?: JSX.Element };
+
+export const Box = createCustomElement<ElementProps<BoxOptions, BoxRenderable> & ContianerProps>("opentui_box", true);
```

Comment:
Is there a reason to make custom capitalised elements as opposed to just having `<box>stuff</box>`?
Having to have to specifically import custom elements from the renderer doesn't really feel natural when you don't really need to. This is one of the biggest limitations of react ink, and we shouldn't need to make the same choice

Reply:
Hmm, yeah hadn't considered. No reason they have to be will try and fix